### PR TITLE
feat: Added Hebrew language to firebase_ui_localizations

### PR DESCRIPTION
**Pull request by beyondeye**
## Description

Added Hebrew language to firebase_ui_localizations

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ x] All existing and new tests are passing.
- [ x] I updated/added relevant documentation (doc comments with `///`).
- [ x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ x] I read and followed the [Flutter Style Guide].
- [ x] I signed the [CLA].
- [ x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x ] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by russellwheatley** [comment]
Hey @beyondeye, do you mind having the translations checked over by someone else who speaks Hebrew with an approval, please? Perhaps a friend/colleague on GitHub? Unfortunately, no maintainers speak Hebrew.

**Comment by beyondeye** [comment]
Hi @russellwheatley. I speak Hebrew myself; I can search also for somebody else for a double check, do you have any specific requirement for the person I should that should check the translation?

**Comment by beyondeye** [comment]
@russellwheatley . So once I have somebody proofread all the translations what will be the procedure next? should I communicate his/her GitHub handle and you will give them authorization  to approve the pull request?

File: packages/firebase_ui_localizations/lib/src/lang/he.dart [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+  @override
+  String get signInMethods => "שיטות כניסה"; //"Sign-in methods";
+
+  @override
+  String get signInText => "כניסה"; //"Sign in";
+
+  @override
+  String get signInWithAppleButtonText =>
+      "התחברות באמצעות חשבון משתמש Apple"; //"Sign in with Apple";
```
Comment (lironhl) [inline packages/firebase_ui_localizations/lib/src/lang/he.dart:211]: In this line and other lines
I prefer account as חשבון and not as חשבון משתמש

File: packages/firebase_ui_localizations/lib/src/lang/he.dart [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+  @override
+  String get signInWithPhoneButtonText =>
+      "כניסה באמצעות מספר טלפון"; //"Sign in with phone";
+
+  @override
+  String get signInWithTwitterButtonText =>
+      "היכנס באמצעות חשבון משתמש בטוויטר"; //"Sign in with Twitter";
+
+  @override
+  String get signOutButtonText => "התנתקות"; //"Sign out";
```
Comment (lironhl) [inline packages/firebase_ui_localizations/lib/src/lang/he.dart:238]: התנתק

File: packages/firebase_ui_localizations/lib/src/lang/he.dart [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+
+  @override
+  String get smsAutoresolutionFailedError =>
+      "הבאת קוד SMS באופן אוטומטי נכשלה. אנא הזן את הקוד שלך באופן ידני"; //"Failed to resolve SMS code automatically. Please enter your code manually";
```
Comment (lironhl) [inline packages/firebase_ui_localizations/lib/src/lang/he.dart:242]: Seems like you have double space.
I'd use ייבוא and not הבאת

File: packages/firebase_ui_localizations/lib/src/lang/he.dart [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+
+  @override
+  String get valueLabel => "ערך"; //"value";
+
+  @override
+  String get verifyCodeButtonText => "אמת"; //"Verify";
+
+  @override
+  String get verifyingSMSCodeText =>
+      "אימות קוד SMS..."; //"Verifying SMS code…";
```
Comment (lironhl) [inline packages/firebase_ui_localizations/lib/src/lang/he.dart:275]: מאמת קוד SMS...

File: packages/firebase_ui_localizations/lib/src/lang/he.dart [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+
+  @override
+  String get chooseACountry => "בחר מדינה"; //"Choose a country";
+
+  @override
+  String get confirmPasswordDoesNotMatchErrorText =>
+      "הסיסמאות אינן תואמות"; //"Passwords do not match";
+
+  @override
+  String get confirmPasswordInputLabel => "אשר סיסמה"; // "Confirm password";
```
Comment (lironhl) [inline packages/firebase_ui_localizations/lib/src/lang/he.dart:31]: וודאי סיסמה?

File: packages/firebase_ui_localizations/lib/src/lang/he.dart [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+
+  @override
+  String get confirmPasswordIsRequiredErrorText =>
+      "אשר את הסיסמה שלך"; //"Confirm your password";
```
Comment (lironhl) [inline packages/firebase_ui_localizations/lib/src/lang/he.dart:35]: וודא את הסיסמה שלך?

File: packages/firebase_ui_localizations/lib/src/lang/he.dart [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+      """הזן את הדוא"ל שלך ואנו נשלח לך קישור לאיפוס הסיסמה"""; //"Provide your email and we will send you a link to reset your password";
+
+  @override
+  String get forgotPasswordViewTitle => "סיסמה נשכחה"; //"Forgotten password";
+
+  @override
+  String get geopointLabel => "גיאופוינט"; //"geopoint";
+
+  @override
+  String get goBackButtonLabel => "חזרה למסך הקודם"; //"Go back";
```
Comment (lironhl) [inline packages/firebase_ui_localizations/lib/src/lang/he.dart:104]: חזרה
In the next translation you only wrote - הבא

File: packages/firebase_ui_localizations/lib/src/lang/he.dart [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+  @override
+  String get phoneNumberIsRequiredErrorText =>
+      "נדרש מספר טלפון"; //"Phone number is required";
+
+  @override
+  String get phoneVerificationViewTitleText =>
+      "הזן את מספר הטלפון שלך"; //"Enter your phone number";
+
+  @override
+  String get profile => "פרופיל משתמש"; //"Profile";
```
Comment (lironhl) [inline packages/firebase_ui_localizations/lib/src/lang/he.dart:173]: פרופיל will be enough

**Comment by russellwheatley** [comment]
Thanks for the input @lironhl 😄. What do you think, @beyondeye?

**Comment by beyondeye** [comment]
@lironhl Thanks a lot! I have committed all the fixes.

File: packages/firebase_ui_localizations/lib/src/lang/he.dart [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+  @override
+  String get confirmPasswordDoesNotMatchErrorText =>
+      "הסיסמאות אינן תואמות"; //"Passwords do not match";
+
+  @override
+  String get confirmPasswordInputLabel => "וודא סיסמה"; // "Confirm password";
+
+  @override
+  String get confirmPasswordIsRequiredErrorText =>
+      "וודא הסיסמה שלך"; //"Confirm your password";
```
Comment (lironhl) [inline packages/firebase_ui_localizations/lib/src/lang/he.dart:35]: וודא את הסיסמה שלך

File: packages/firebase_ui_localizations/lib/src/lang/he.dart [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+  String get signInWithGoogleButtonText =>
+      "כניסה באמצעות חשבון של Google"; //"Sign in with Google";
+
+  @override
+  String get signInWithPhoneButtonText =>
+      "כניסה באמצעות מספר טלפון"; //"Sign in with phone";
+
+  @override
+  String get signInWithTwitterButtonText =>
+      "היכנס באמצעות חשבון בטוויטר"; //"Sign in with Twitter";
```
Comment (lironhl) [inline packages/firebase_ui_localizations/lib/src/lang/he.dart:235]: כניסה באמצעות חשבון טוויטר 
Like the others

File: packages/firebase_ui_localizations/lib/src/lang/he.dart [hunk 11]
```diff
@@ -0,0 +1,289 @@
...
+
+  @override
+  String get signOutButtonText => "התנתק"; //"Sign out";
+
+  @override
+  String get smsAutoresolutionFailedError =>
+      "ייבוא קוד SMS באופן אוטומטי נכשלה. אנא הזן את הקוד שלך באופן ידני"; //"Failed to resolve SMS code automatically. Please enter your code manually";
```
Comment (lironhl) [inline packages/firebase_ui_localizations/lib/src/lang/he.dart:242]: ייבוא קוד ה-SMS באופן אוטומטי נכשל. אנא הזן את הקוד שלך באופן ידני

**Review by lironhl — CHANGES_REQUESTED** [review]
Sorry, but couple of comments and you're good to go

**Comment by beyondeye** [comment]
@lironhl Again thank you. I hope that now we hunted down all the errors.

**Review by lironhl — APPROVED** [review]

**Review by lironhl — APPROVED** [review]

**Review by russellwheatley — APPROVED** [review]

**Review by lesnitsky — APPROVED** [review]

**Pull request merged**